### ISSUE TITLE
Apply gradient accumulation fix to DPO/PPO recipes

### ISSUE THREAD

**Issue by bogdansalyp** [pytorch/torchtune#2334]
https://unsloth.ai/blog/gradient

**Comment by bogdansalyp** [comment]
Depends on #2335

**Comment by bogdansalyp** [comment]
SFT fix was addressed here: https://github.com/pytorch/torchtune/pull/1875
